pkg/apis/migration/v1alpha1/verify.go

```diff
@@ -0,0 +1,208 @@
+type Workload struct {
+	Name      string   `json:"name"`
+	Resources []string `json:"resources,omitempty"`
+}
+
+// UnhealthyNamespace is a store for unhealthy resources in a namespace
+type UnhealthyNamespace struct {
+	Namespace string     `json:"namespace"`
+	Workloads []Workload `json:"workloads"`
+}
+
+// VerifyPods would collect, display and notify about the pod health in a specific namespace
+func (u *UnhealthyResources) VerifyPods(client k8sclient.Client, options *k8sclient.ListOptions) (*[]unstructured.Unstructured, error) {
+	podList := corev1.PodList{}
+
+	err := client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return nil, err
+	}
+
+	unhealthy := []unstructured.Unstructured{}
+	for _, pod := range podList.Items {
+		switch pod.Status.Phase {
+		case corev1.PodFailed:
+		case corev1.PodRunning:
+			if !containerUnhealthy(pod) {
+				continue
+			}
+		default:
+			continue
+		}
+
+		obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&pod)
+		if err != nil {
+			return nil, err
+		}
+
+		unstructuredPod := unstructured.Unstructured{Object: obj}
+		unstructuredPod.SetKind("Pod")
+		unhealthy = append(unhealthy, unstructuredPod)
+	}
+
+	return &unhealthy, nil
+}
+
+func containerUnhealthy(pod corev1.Pod) bool {
+	// Consider suspicious containers as an error state
+	podContainersStatuses := append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...)
+	for _, containerStatus := range podContainersStatuses {
+		if containerStatus.Ready {
+			continue
+		}
+		terminationDetails := containerStatus.State.Terminated
+		if terminationDetails == nil || terminationDetails.ExitCode == 0 {
+			continue
+		}
+		return true
+	}
+
+	return false
+}
+
+func (u *UnhealthyResources) findResources(cluster *[]UnhealthyNamespace, namespace string) *UnhealthyNamespace {
+	for _, resource := range *cluster {
+		if resource.Namespace == namespace {
+			return &resource
+		}
+	}
+	return nil
+}
+
+func (u *UnhealthyResources) findWorkload(resources *UnhealthyNamespace, name string) *Workload {
+	for _, workload := range resources.Workloads {
+		if workload.Name == name {
+			return &workload
+		}
+	}
+	return nil
+}
+
+// AddWorkload is adding a workload to selected list of unhealthy namespaces
+func (u *UnhealthyResources) AddWorkload(cluster *[]UnhealthyNamespace, namespace, name string, resourceNames ...string) {
+	workload := Workload{
+		Name:      name,
+		Resources: resourceNames,
+	}
+	resources := u.findResources(cluster, namespace)
+	if resources == nil {
+		unhealthyNamespace := UnhealthyNamespace{
+			Namespace: namespace,
+			Workloads: []Workload{workload},
+		}
+		*cluster = append(*cluster, unhealthyNamespace)
+	} else {
+		resources.Workloads = append(resources.Workloads, workload)
+	}
+}
+
+// AddUnhealthyResources is adding unhealthy workloads and resolves owner references
+func (u *UnhealthyResources) AddUnhealthyResources(
+	client k8sclient.Client,
+	targetCluster *[]UnhealthyNamespace,
```

Comment:
Consider a better name for this. Even `unhealthyNamespaces` would suffice. `targetCluster` makes me think the object is a client or something.